docs/2.6.0/docs/deploy-daml/infrastructure-architecture/high-availability/basic-ha/participant.rst

```diff
@@ -1,142 +1,94 @@
-
-Ledger API Client Fail-Over via Load Balancer
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-To hide the fact that a participant is replicated and to offer a single ledger
-API endpoint towards applications, we recommend the usage of layer 4 (=TCP
-level), highly available load balancer.
-
-The load balancer (LB) is configured with a pool of backend servers based on the
-ledger API server addresses and ports of the participant node replicas. The
-participant node replicas expose their status if they are the active or passive
-replica via a health endpoint. The LB periodically checks the health API
-endpoint of the replicas and marks a backend server offline if the replica is
-passive. Thus the load balancer only sends requests to the active backend
-server. The polling frequency of the health endpoints affect the fail-over
-times.
-
-During fail-over requests may still be send to the former active replica, which
-will be rejected and the application has to retry the submission of commands in
-that case until they are forwarded to the new active replica.
+Exclusive application level locks are identified by a 30 bit integer lock id which is allocated based on a scope name and counter. 
+
+The lock counter differentiates locks used in Canton from each other, depending on their usage. The scope name ensures the uniqueness of the lock id for a given lock counter. The allocation process generates a unique lock id by hashing and truncating the scope and counter to 30 bits.
+
+.. NOTE::
+  On Oracle, the lock scope is the schema name, i.e. user name. On PostgreSQL, it is the name of the database. 
+  
+Participant replicas must allocate lock ids and counters consistently. It is, therefore, crucial that replicas are configured with the same storage configuration, e.g. for Oracle using the correct username to allocate the lock ids within the correct scope.
+
+Prevent Passive Replica Activity
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. IMPORTANT::
+  Passive replicas do not hold the exclusive lock and cannot write to the shared database. 
+
+To avoid passive replicas attempting to write to the database - any such attempt fails and produces an error - we use a coarse-grained guard on domain connectivity and API services.
+
+To prevent the passive replica from processing domain events, and ensure it rejects incoming Ledger API requests, we keep the passive replica disconnected from the domains as a coarse-grained enforcement.
+
+Lock Loss and Failover
+""""""""""""""""""""""
+
+If the active replica crashes or loses connection to the database, the lock is released and a passive replica can claim the lock and become active. Any pending writes in the formerly active replica fail due to losing the underlying connection and the corresponding lock.
+
+The active replica has a grace period in which it may rebuild the connection and reclaim the lock to avoid unnecessary failover on short connection interruptions. 
```

Comment:
slightly outdated: the grace period now comes from the higher frequency of checking the health of the lock in the active replica vs. trying to acquire the lock in the passive replica at a lower frequency.